management/dns_update.py

```diff
@@ -303,6 +303,22 @@ def has_rec(qname, rtype, prefix=None):
 		if not has_rec(qname, rtype):
 			records.append((qname, rtype, value, explanation))
 
+	# Adds autoconfiguration A records for all domains.
+	# mta-sts.* - required A record for mta-sts (serving the policy)
+	mta_sts_records = [
+		("mta-sts", "A", env["PUBLIC_IP"], "Provides MTA-STS support"),
+		("mta-sts", "AAAA", env.get('PUBLIC_IPV6'), "Provides MTA-STS support"),
+		("_mta-sts", "TXT", "v=STSv1; id=%sZ" % datetime.datetime.now().strftime("%Y%m%d%H%M%S"), "Enables MTA-STS support")
+	]
+	# Skip if the user has set a custom _smtp._tls record.
+	if not has_rec("_smtp._tls", "TXT", prefix="v=TLSRPTv1;"):
+		mta_sts_records.append(("_smtp._tls",  "TXT", "v=TLSRPTv1;", "For reporting, add an mail alias, for example 'tlsrpt@%s' and a custom TXT record like 'v=TLSRPTv1; rua=mailto:tlsrpt@%s' for reporting" % (env["PRIMARY_HOSTNAME"], env["PRIMARY_HOSTNAME"]) ))
```

Comment:
What's better. Always having a  report to postmaster@PRIMARY_HOSTNAME or enabling reporting to the user if he creates tlsrpt@PRIMARY_HOSTNAME?

Reply:
jvolkenant, i added comments to web.sh;

if the alias (or mailbox) tlsrpt@PRIMARY_HOSTNAME exists, a reporting rule is added v=TLSRPTv1; rua=mailto:tlsrpt@PRIMARY_HOSTNAME

If this alias is not available, only v=TLSRPTv1 is added. It's up to the user to change this to a desired email address. 

I didn't use postmaster for this, because this kind of emails will probably be processed by an external party.